redux/react/ConnectMacro.hx

```diff
@@ -81,11 +87,78 @@ class ConnectMacro
 
 	/* MAP STATE */
 
-	static function hasMapState(fields:Array<Field>)
+	static function getMapState(fields:Array<Field>)
 	{
 		for (field in fields)
-			if (field.name == 'mapState') return true;
-		return false;
+			if (field.name == 'mapState') return field;
+		return null;
+	}
+
+	static function checkMapStateType(mapStateField:Field, inClass:ClassType)
+	{
+		var propsType:Type = TDynamic(null);
+		var stateType:Type = TDynamic(null);
+
+		switch (inClass.superClass)
+		{
+			case {params: params, t: _.toString() => 'react.ReactComponentOf'}:
+				propsType = params[0];
+				stateType = params[1];
+
+			default:
+		}
+
+		switch (mapStateField.kind)
+		{
+			case FFun(f):
+				if (f.args.length != 2)
```

Comment:
Should we accept `mapState(state)` without `ownProps`?

Reply:
Hmm, I guess we could :)